flathunter/crawler/immobilienscout.py

```diff
@@ -181,12 +181,26 @@ def get_page(self, search_url, driver=None, page_no=None):
 
     def get_expose_details(self, expose):
         """Loads additional details for an expose by processing the expose detail URL"""
-        soup = self.get_soup_from_url(expose['url'])
+        driver = self.get_driver()
+        if driver is not None:
```

Comment:
So do I understand from this that you need to open a second tab to load the expose to get the details? Is there a reason to do this here instead of in the `OpenAIProcessor`? If we have to do it here, can we have a switch to skip this if the OpenAI feature is disabled? Otherwise everybody not using OpenAI will make a bunch of unnecessary fetches.

Reply:
Actually, the whole `get_expose_details` is broken for me with immobilienscout if one does not add a driver. This is due to a request initially returning "Checking that you're not a robot" and only once this check is finished the final content of the page will be returned. But I can simply cherry-pick this into a different pull request.